base_event.cpp

```diff
@@ -1,92 +1,153 @@
-  wxFileInputStream input_stream(openFileDialog.GetPath());
-  // if filepath is invalid
-  if (!input_stream.IsOk())
-  {
-    wxLogError("Cannot open file '%s'.", openFileDialog.GetPath());
-    return;
-  }
-  textbox->SetLabel(openFileDialog.GetPath());
-  return;
-}
+#include <wx/wxprec.h>
+
+#ifndef WX_PRECOMP
+#  include <wx/wx.h>
+#endif
+
+#include "base.h"
+#include "controller.h"
+#include <vector>
+#include <utility>
+
+/////////////////
+// EVENT TABLE //
+/////////////////
+
+BEGIN_EVENT_TABLE(MainFrame, wxFrame)
+  EVT_BUTTON(BUTTON_Calc, MainFrame::OnCalc)
+  EVT_BUTTON(BUTTON_Browse, MainFrame::OnAtomBrowse)
+  EVT_BUTTON(BUTTON_Radius, MainFrame::OnRadiusBrowse)
+  EVT_BUTTON(BUTTON_LoadFiles, MainFrame::OnLoadFiles)
+  EVT_GRID_CELL_CHANGING(MainFrame::GridChange)
+END_EVENT_TABLE()
+
+////////////////////////////////
+// METHODS FOR EVENT HANDLING //
+////////////////////////////////
+
+// exit program
+void MainFrame::OnExit(wxCommandEvent& event)
+{
+  this->Close(TRUE);
+}
+
+// display text in output, used for debugging
+void MainFrame::OnPrint(wxCommandEvent& event)
+{
+  std::string text = "treat yourself well";
+  printToOutput(text);
+  return;
+}
+
+// begin calculation
+void MainFrame::OnCalc(wxCommandEvent& event){
+
+  //std::string filepath = (filepathText->GetValue()).ToStdString();
+  //  printToOutput(filepath);
+
+  // Deactivate the Start and Load files buttons during the calculation
+  MainFrame::calcButton->Enable(false);
+  MainFrame::loadFilesButton->Enable(false);
+  // Prevents editing of the atom list grid during the calculation
+  MainFrame::atomListGrid->EnableEditing(false);
+  // so far only xyz files allowed
+  Ctrl::getInstance()->runCalculation();
+  // without wxYield, the button is grayed but still records clicks
+  // yet, wxYield is apparently dangerous in an event handler, need to find an alternative
+  wxYield();
+  // Reactivate the Start and Load files buttons and the atom list edition
+  MainFrame::calcButton->Enable(true);
+  MainFrame::loadFilesButton->Enable(true);
+  MainFrame::atomListGrid->EnableEditing(true);
+  return;
+}
+
+// browse for atom file
+void MainFrame::OnAtomBrowse(wxCommandEvent& event){
+  std::string filetype = "XYZ files (*.xyz)|*.xyz";
+  OnBrowse(filetype, filepathText);
+  return;
+}
+
+// browse for radius file
+void MainFrame::OnRadiusBrowse(wxCommandEvent& event){
+  std::string filetype = "TXT files (*.txt)|*.txt";
+  OnBrowse(filetype, radiuspathText);
+  return;
+}
+
+// load input files to display radii list
+void MainFrame::OnLoadFiles(wxCommandEvent& event){
+
+  // Deactivate the Start and Load files buttons during the loading
+  MainFrame::calcButton->Enable(false);
+  MainFrame::loadFilesButton->Enable(false);
+  // so far only xyz files allowed
+  std::vector<std::tuple<std::string, int, double>> atoms_for_list = Ctrl::getInstance()->loadInputFiles();
+  MainFrame::generateAtomList(atoms_for_list);

```

Comment:
I have an issue with this. This event function should really only call the Controller function `loadInputFiles()`. The rest should be handled by the controller, not by this function. This means that the controller should generate the vector of tuples, then call `generateAtomList()` directly from the `loadInputFiles()` function.

Reply:
I suppose the limit between what should be managed by the gui, the controller and the back-end model isn't crystal clear to me (I am at fault).
It just seemed so intricate to have all these extra communications steps between front-end and back-end through the controller while information could be generated and stored where it will be used directly.